src/main/kotlin/app/shosetsu/lib/lua/ShosetsuLuaLib.kt

```diff
@@ -242,6 +243,17 @@ class ShosetsuLuaLib : TwoArgFunction() {
 		fun Log(name: String, arguments: String) {
 			ShosetsuSharedLib.logger(name, arguments)
 		}
+
+		/**
+		 * Delay by [milliseconds]
+		 */
+		fun delay(milliseconds: Long) {
+			val startTime = Clock.System.now().toEpochMilliseconds()
+			var currentTime: Long
+			do {
+				currentTime = Clock.System.now().toEpochMilliseconds()
+			} while (startTime + milliseconds > currentTime)
```

Comment:
Timers using infinite loops are inefficient and waste CPU time. We should figure out a better way to make a delay function. Furthermore, I don't know if such a function is even needed right now.

Reply:
The usage of such a function is to allow scripts to not create network calls too quickly. As this often will lead to website blocking users.

The loop overhead is only 23~ ms on my desktop, and shouldn't be too much more on mobile devices.

Note: Not an infinite loop, as there is a clear end clause.

Reply:
@TechnoJo4 Swapping to use kotlin-corutines for the delay call leads to a 62ms overhead. Currently this is far more performant then the other multi platform method

Reply:
A 60ms difference is insignificant, especially when waiting for e.g. a rate limit. A web request might take much longer.

Reply:
So, Is this acceptable to merge or?